R/datamapping-xygroup.R

```diff
@@ -12,12 +12,12 @@ XYGDataMapping <- R6::R6Class(
       super$initialize(...)
       self$groupMapping <- groupMapping %||% GroupMapping$new()
     },
-    
+
     checkMapData = function(data, metaData = NULL) {
       validateMapping(self$x, data)
-      validateMapping(self$y, data)
+      validateMapping(self$y, data, nullAllowed = TRUE)
 
-      self$data <- data[, c(self$x, self$y)]
+      self$data <- data[, c(self$x, self$y), drop = FALSE]

```

Comment:
This is typically the kind of changes that require an inline comment. This is absolutely not clear why you use drop in this case and not in any other case. We should not comment the obvious, but we should absolutely comment this types of implementation